src/lib.rs

```diff
@@ -1470,16 +1470,25 @@ pub async fn run_tests_parallel<'a>(
                         }
                     }
                     JobEvent::NewJob(job) => {
-                        pending_jobs.push_back(job);
-                        progress_bar.inc_length(1);
-                        if progress_bar.is_hidden() {
-                            print_progress(
-                                logger,
-                                start_instant,
-                                &mut last_progress_print,
-                                job_id - job_executor.len() as u64,
-                                job_id + pending_jobs.len() as u64,
-                            );
+                        let mut bisect_finished = false;
+                        if let Job::Bisect { list, .. } = &job {
+                            if list.len() <= 2 {
+                                trace!(logger, "Bisect succeeded, two tests or less left");
+                                bisect_finished = true;
+                            }
+                        }
+                        if !bisect_finished {
+                            pending_jobs.push_back(job);
+                            progress_bar.inc_length(1);
```

Comment:
nit: Please add a comment that all new jobs return exactly one test result, so increasing the size by one here is correct.
You can also add an `assert!(!matches!(job, Job::FirstRun), "Unexpected FirstRun")`